plonetheme/onegovbear/browser/dynamic_scss_resources.py

```diff
@@ -0,0 +1,59 @@
+from Acquisition._Acquisition import aq_inner
+from Acquisition._Acquisition import aq_parent
+from Products.CMFPlone.interfaces import IPloneSiteRoot
+from ftw.theming.interfaces import ISCSSResourceFactory
+from ftw.theming.resource import SCSSResource
+from plone.app.layout.navigation.interfaces import INavigationRoot
+from plonetheme.onegovbear.browser.forms import ANNOTATION_KEY
+from zope.annotation import IAnnotations
+from zope.interface import implements
+
+
+class CustomVariablesResourceFactory(object):
+    implements(ISCSSResourceFactory)
+    annotations_key = ANNOTATION_KEY
+
+    def __call__(self, context, request):
+        self.context = context
+        self.request = request
+        return self.get_resource()
+
+    def get_resource(self):
+        source = self.get_source()
+        return SCSSResource(ANNOTATION_KEY, slot='variables', source=source)
+
+    def get_source(self):
+        ancestor_variables = self.get_ancestor_variables()
+
+        variables = {}
+        for d in ancestor_variables:
+            variables.update(d)
+
+        source = ''
+        if variables:
```

Comment:
I'd just return when variables is empty, e.g.:
```python
if not variables:
  return ''

return '; '.join(.....)
```

See also http://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html